src/seeds.gratitude.cpp

```diff
@@ -0,0 +1,197 @@
+#include <seeds.gratitude.hpp>
+
+
+ACTION gratitude::reset () {
+  require_auth(get_self());
+
+  auto bitr = balances.begin();
+  while (bitr != balances.end()) {
+    bitr = balances.erase(bitr);
+  }
+
+  auto sitr = sizes.begin();
+  while (sitr != sizes.end()) {
+    sitr = sizes.erase(sitr);
+  }
+
+  auto stitr = stats.begin();
+  while (stitr != stats.end()) {
+    stitr = stats.erase(stitr);
+  }
+
+  // setup first round
+  stats.emplace(_self, [&](auto& item) {
+    item.round_id = 1;
+    item.num_transfers = 0;
+    item.volume = asset(0, gratitude_symbol);
+  });
+
+}
+
+ACTION gratitude::give (name from, name to, asset quantity, string memo) {
+  require_auth(from);
+
+  check( from != to, "gratitude: cannot give to self" );
+  check( is_account( to ), "gratitude: to account does not exist");
+  check_asset(quantity);
+  check( quantity.amount > 0, "gratitude: must give positive quantity" );
+
+  // Should create balances if not there yet
+  init_balances(to);
+  init_balances(from);
+
+  sub_gratitude(from, quantity);
+  add_gratitude(to, quantity);
+
+  update_stats(from, to, quantity);
+}
+
+ACTION gratitude::newround() {
+  require_auth(get_self());
+  auto generated_gratz = config_get(gratzgen);
+
+  auto contract_balance = eosio::token::get_balance(contracts::token, get_self(), seeds_symbol.code());
+  uint64_t tot_accounts = get_size("balances.sz"_n);
+  uint64_t volume = get_current_volume();
+
+
+  auto balances_by_received = balances.get_index<"byreceived"_n>();
+
+  auto bitr = balances_by_received.begin();
+  while (bitr != balances_by_received.end()) {
```

Comment:
This will only work up to a balances size of about 200 - 400, past that it won't work

Which may not be an issue- certainly won't run into this any time soon if we use this inside hypha.

Reply:
Well, in `newround` we don't need to iterate by received, anyway, since we're paying every account with higher than 0 balance.